docs/src/main/asciidoc/includes/tracing/common-callbacks.adoc

```diff
@@ -0,0 +1,159 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2024 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+ifndef::rootdir[:rootdir: {docdir}/../..]
+include::{rootdir}/attributes.adoc
+
+// tag::defs[]
+:tracing-javadoc: {tracing-javadoc-base-url}/io/helidon/tracing
+:tracing-api-javadoc: {tracing-javadoc}
+:listener-type: SpanLifeCycleListener
+// end::defs[]
+
+
+// tag::detailed[]
+// tag::intro[]
+== Responding to Span Life Cycle Events [[Tracing-callbacks]]
+
+Applications and libraries can register listeners to be notified at several moments during the lifecycle of every span:
+
+* Before a new span starts
+* After a new span has started
+* After a span ends
+* After a span is activated (creating a new scope)
+* After a scope is closed
+
+// end::intro[]
+
+The next sections explain how you can write and add a listener and what it can do. See the link:{tracing-api-javadoc}/{listener-type}.html[`{listener-type}`] JavaDoc for more information.
+
+=== Understanding What Listeners Do
+A listener cannot affect the lifecycle of a span or scope it is notified about, but it can add tags and events and update the baggage associated with a span.
+Often a listener does additional work that does not change the span or scope such as logging a message.
+
+When Helidon invokes the listener's methods it passes safe adapters for the parameter types. These adapters limit the access the listener has to the span builder, span, or scope, as summarized in the following table.
```

Comment:
"safe adapters" _I_ know what you mean but perhaps using the word "proxy" in some way would help others understand.

Reply:
Changed to "proxy/proxies."